### PR TITLE
Add Rust Docs examples for UnionArray

### LINKED ISSUE #751: Add examples to Rust Docs for arrow::array::UnionArray
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I would like to see examples of how to use UnionArrays in the Rust docs (https://docs.rs/arrow/5.3.0/arrow/array/struct.UnionArray.html).  The explanations are good, but examples would improve my understanding.

**Describe the solution you'd like**
Add examples for `new` and try_new`

**Describe alternatives you've considered**

**Additional context**
I thought of this while I was looking into #654

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

# Which issue does this PR close?

Closes #751 

# Rationale for this change
 
 
Add 2 docs examples of building `UnionArray` using the function `UnionArray::try_new`,
so that users can easily understand the data structure of `UnionArray`.
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by tustvold — COMMENTED** [review]
This looks good to me :+1:

File: arrow/src/array/array_union.rs [hunk 1]
```diff
@@ -35,6 +35,73 @@ use std::any::Any;
...
+/// ];
+///
+/// let array = UnionArray::try_new(
+///     type_id_buffer,
+///     None,
+///     children,
+///     None,
+/// ).unwrap();
+///
+/// let value = array.value(0).as_any().downcast_ref::<Int32Array>().unwrap().value(0);
```
Comment (alamb) [inline arrow/src/array/array_union.rs:96]: It isn't the nicest API in the world, is it ?

However, documenting what we have is 👍 

Thank you @HaoYang670

**Review by alamb — APPROVED** [review]
Thank you very much for the contribution @HaoYang670

**Comment by matthewmturner** [comment]
thank you @HaoYang670

**Pull request merged**